app/controllers/api/v1/teachers_controller.rb

```diff
@@ -103,6 +103,71 @@ def name_valid
     end
   end
 
+  def get_enews_subscription
+    teacher_id = params.require(:id)
```

Comment:
This will need some authorization added to it. If the teacher is always the one requesting this, then we can require that he current_user is an admin or the teacher.  Pundit should be used for this authorization.